question21/main.js

```diff
@@ -0,0 +1,162 @@
+  userId: "ID",
+  name: "名前",
+  gender: "性別",
+  age: "年齢"
+}
+
+const renderTableElement = (userData)  => {
+  const table = createElementWithClassName("table", "table");
+  const tbody = createElementWithClassName("tbody", "tbody");
+  table.id = "js-table";
+  tbody.id = "js-tbody";
+  const tableHeadValue = Object.values(userDataColumn);
+  wrapper.appendChild(table).appendChild(getTableHeadElement(tableHeadValue));
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableRowFragment(userData));
+  sortButtonClickEvent(userData);
+}
+
+const getTableHeadElement = (tableHeadValue) => {
+  const thead = createElementWithClassName("thead", "thead");
+  const tr = createElementWithClassName("tr", "tr");
+  const fragmentThElement = document.createDocumentFragment();
+  for (const value of tableHeadValue) {
+    const th = createElementWithClassName("th", "th");
+    th.textContent = value;
+    value === 'ID' && th.appendChild(createSortButton());
+    fragmentThElement.appendChild(tr).appendChild(th);
+  }
+  thead.appendChild(fragmentThElement);
+  return thead;
+}
+
+const getTableRowFragment = (userData) => {
+  const tableRowfragment = document.createDocumentFragment();
+  for (let i = 0; i < userData.length; i++) {
+    const tr = createElementWithClassName("tr", "tr");
+    tableRowfragment.appendChild(tr).appendChild(getTableDataFragment(userData[i]));
+  }
+  return tableRowfragment;
+}
+
+const getTableDataFragment = (userData) => {
+  const tableDatafragment = document.createDocumentFragment();
+  Object.keys(userDataColumn).forEach((key) => {
+    const td = createElementWithClassName("td", "td");
+    td.textContent = userData[key];
+    tableDatafragment.appendChild(td);
+  });
+  return tableDatafragment;
+};
+
+const createSortButton = () => {
+  const sortButton = createElementWithClassName("button", "sortButton");
+  sortButton.dataset.status = "default";
+  sortButton.id = "js-sortbtn";
+  return sortButton;
+}
+
+const sortButtonClickEvent = () => {
+  const tbody = document.querySelector('tbody');
+  const sortButton = document.getElementById('js-sortbtn');
+  const trElement = [...document.querySelectorAll("tbody > tr")];
+  sortButton.addEventListener('click', (e) => {
+    const clickedCellIndex = e.target.parentElement.cellIndex;
+    const nextStatus = switchSortStatus(e.target);
+    sortButton.dataset.status = nextStatus;
+    const sortedRows = sortFunc(e.target,clickedCellIndex,trElement);
+    while (tbody.firstChild) {
+      tbody.removeChild(tbody.firstChild);
+  }
+    sortedRows.forEach((row) => {
+      tbody.appendChild(row);
+    });
+  });
+}
+
+const Sort = {
+  Default: "default",
+  Asc: "asc",
+  Desc: "desc"
+};
+
+const switchSortStatus = (target) => {
+  switch (target.dataset.status) {
+    case Sort.Default:
+      return Sort.Asc;
+    case Sort.Asc:
+      return Sort.Desc;
+    case Sort.Desc:
+      return Sort.Default; 
+    default:
+      return Sort.defaul;
```

Comment:
Fix this part and it's perfect!
```suggestion
      return Sort.default;
```

Reply:
What happened here!!?:scream: Thank you for pointing it out and fixed!